src/opm/io/eclipse/EGrid.cpp

```diff
@@ -355,4 +355,188 @@ void EGrid::getCellCorners(int globindex, std::array<double,8>& X,
     return getCellCorners(ijk_from_global_index(globindex),X,Y,Z);
 }
 
+
+std::vector<std::array<float, 3>> EGrid::getXYZ_layer(int layer, std::array<int, 4>& box, bool bottom)
+{
+   // layer is layer index, zero based. The box array is i and j range (i1,i2,j1,j2), also zero based
+
+    if ((layer < 0) || (layer > (nijk[2] -1))){
+        std::string message = "invalid layer index " + std::to_string(layer) + ". Valied range [0, ";
+        message = message + std::to_string(nijk[2] -1) + "]";
+        throw std::invalid_argument(message);
+    }
+
+    if ((box[0] < 0) || (box[0]+1 > nijk[0]) || (box[1] < 0) || (box[1]+1 > nijk[0]) ||
+         (box[2] < 0) || (box[2]+1 > nijk[1]) || (box[3] < 0) || (box[3]+1 > nijk[1]) ||
+           (box[0] > box[1]) || (box[2] > box[3])){
+
+        throw std::invalid_argument("invalid box input, i1,i2,j1 or j2 out of valied range ");
+    }
+
+    int nodes_pr_surf = nijk[0]*nijk[1]*4;
+    int zcorn_offset = nodes_pr_surf * layer * 2;
+
+    if (bottom)
+        zcorn_offset += nodes_pr_surf;
+
+    std::vector<float> layer_zcorn;
+    layer_zcorn.reserve(nodes_pr_surf);
+
+    std::vector<std::array<float, 3>> xyz_vector;
+
+    if (coord_array.size() == 0)
+        coord_array = getImpl(coord_array_index, REAL, real_array, "float");
+
+    if (zcorn_array.size() > 0){
+        for (size_t n = 0; n < static_cast<size_t>(nodes_pr_surf); n++)
+            layer_zcorn.push_back(zcorn_array[zcorn_offset + n]);
+
+    } else {
+        layer_zcorn = get_zcorn_from_disk(layer, bottom);
+    }
+
+    std::array<double,4> X;
+    std::array<double,4> Y;
+    std::array<double,4> Z;
+
+    std::array<int,3> ijk;
+    ijk[2]=0;
+
+    for (int j = box[2]; j < (box[3] + 1); j++) {
+        for (int i = box[0]; i < (box[1] + 1); i++) {
+
+            ijk[0]=i;
+            ijk[1]=j;
+
+            this->getCellCorners(ijk, layer_zcorn, X, Y, Z );
+
+            for (size_t n = 0; n < 4; n++){
+                std::array<float, 3> xyz;
+                xyz[0] = X[n];
+                xyz[1] = Y[n];
+                xyz[2] = Z[n];
+
+                xyz_vector.push_back(xyz);
+            }
+        }
+    }
+
+    return xyz_vector;
+}
+
+
+std::vector<std::array<float, 3>> EGrid::getXYZ_layer(int layer, bool bottom)
+{
+    std::array<int, 4> box = {0, nijk[0] -1 , 0, nijk[1] -1 };
+    return this->getXYZ_layer(layer, box, bottom);
+}
+
+std::vector<float> EGrid::get_zcorn_from_disk(int layer, bool bottom)
+{
+    std::vector<float> zcorn_layer;
+    std::fstream fileH;
+
+    int nodes_pr_surf = nijk[0]*nijk[1]*4;
+    int zcorn_offset = nodes_pr_surf * layer * 2;
+
+    if (bottom)
+        zcorn_offset+=nodes_pr_surf;
+
+    fileH.open(inputFileName, std::ios::in |  std::ios::binary);
+
+    if (!fileH)
+        throw std::runtime_error("Can not open EclFile: {}" + this->inputFilename);
```

Comment:
You'll need to use `mt::format()` here if you want to replace the braces (`{}`) with the filename:
```C++
throw std::runtime_error(fmt::format("Unable to open grid input file {}", this->inputFilename));
```
In that case you'll also need to
```C++
#include <fmt/format.h>
```
Technically you should also
```C++
#include <stdexcept>
```
in order to bring a declaration of `std::runtime_error` into scope.

Reply:
This was copied from a different source, and I forgot to remove the braces ({}). I would prefere not to use fmt

``` c++
    if (!fileH)
        throw std::runtime_error("Can not open EGrid file" + this->inputFilename);
```

I also have added 

```c++
#include <stdexcept>
```